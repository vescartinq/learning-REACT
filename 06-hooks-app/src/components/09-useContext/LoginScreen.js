import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  let user = {
    id: 123,
    name: 'Victor',
  };

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(user)}>
        Login
      </button>
    </div>
  );
};
