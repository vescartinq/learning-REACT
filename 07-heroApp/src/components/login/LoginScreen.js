import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { actionTypes } from '../../types/actionTypes';

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext);

  const handleLogin = () => {
    // recupera la última ubicación despues de loggout -> lastPath debe ser igual que el indicado en PrivateRoute.js
    const lastPath = localStorage.getItem('lastPath') || '/';

    dispatch({
      type: actionTypes.LOGIN,
      payload: {
        name: 'Victor',
      },
    });

    history.replace(lastPath);
    // history.push('/'); // redirecciona a la dirección que indiquemos
    // history.replace('/'); // no visita el login, va directo al Dashboard
    // Lo utilizamos en esta App para redirect a /marvel
  };

  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />
      <button className="btn btn-primary" onClick={handleLogin}>
        Login
      </button>
    </div>
  );
};
