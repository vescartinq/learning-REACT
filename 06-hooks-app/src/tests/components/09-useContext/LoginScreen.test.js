import React from 'react';
import { mount } from 'enzyme';
import { UserContext } from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';

describe('Tests in <LoginScreen />', () => {
  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider
      value={{
        setUser,
      }}
    >
      <LoginScreen />
    </UserContext.Provider>
  );

  test('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should execute setUser with the expected argument', () => {
    // Simular onClick del Button
    wrapper.find('button').prop('onClick')();

    expect(setUser).toHaveBeenCalledWith({
      id: 123,
      name: 'Victor',
    });
  });
});
