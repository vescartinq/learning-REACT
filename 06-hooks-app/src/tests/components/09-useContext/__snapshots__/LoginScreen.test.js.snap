// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <LoginScreen /> should render component 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
